docs/developing_iree/profiling_with_tracy.md

```diff
@@ -0,0 +1,149 @@
+# Profiling with Tracy
+
+[Tracy](https://github.com/wolfpld/tracy) is a profiler that puts together in a single view:
+* Both CPU and GPU profiling.
+* Both sampling and instrumentation.
+* Both specifics of our own process, and whole-system profiling a la "systrace".
+
+Since Tracy relies on instrumentation, it requires IREE binaries to be built with a special flag to enable it.
+
+There are two components to Tracy. They communicate over a TCP socket.
+* The "client" is the program being profiled.
+* The "server" is the Tracy profiler UI.
+
+## The Tracy manual
+
+The primary source of Tracy documentation, including for build instructions, is a PDF manual that's part of each numbered release. To find the latest one, navigate [here](https://github.com/wolfpld/tracy/releases) and search for `tracy.pdf`.
+
+## Building the Tracy UI (the "server")
+
+This is explained in section 2.3 of the manual for Windows and Linux. Here we give some more detailed instructions for some systems.
+
+In any case, start by checking out the Tracy repository:
+```
+git clone https://github.com/wolfpld/tracy.git
+```
+
+Note: the IREE repository contains its own clone of the Tracy repository in `third_party/tracy`. You can use it directly here if you prefer, but as `make` does not support out-of-tree builds, this will create the build files into your IREE clone. If you go down this route, consider expanding your `.gitignore`.
```

Comment:
`.git/info/exclude` is also an option: https://docs.github.com/en/free-pro-team@latest/github/using-git/ignoring-files#excluding-local-files-without-creating-a-gitignore-file

Reply:
I've changed the recommended approach: now recommending to build in place in third_party/tracy - like the old doc.
I realized I was worrying for no good reason. Due to how git submodules work, no gitignore is needed.